### DISCUSSION TITLE
Cannot change auth endpoints in custom plugin

### DISCUSSION THREAD

**Discussion by NMFES** [nuxt-community/auth-module#1517] [Support Requests]
### Version

[5](https://github.com/auth-module/releases/tag/5)

### Reproduction link

[-](-)

### Steps to reproduce

nuxt.config.js
================

```
modules: [
        '@nuxtjs/auth-next',
]

auth: {
        strategies: {
            'laravelJWT': {
                provider: 'laravel/jwt',
                // NO HARDCODED ENDPOINTS HERE
            },
        },
        plugins: ['~/plugins/auth.js'],
    },
```

plugins/auth.js
================

```
export default function ({app, $auth}) {
    const url = 'http://api.game.com';

    $auth.strategies.laravelJWT.options.endpoints = {
        login: {url: url + '/auth/login', method: 'post'},
        refresh: {url: url + '/auth/refresh', method: 'post'},
        logout: {url: url + '/auth/logout', method: 'post'},
        user: {url: url + '/auth/user', method: 'get'}
    };
}
```

If we try to log in - it works correctly. But once we have logged in and trying to refresh the page - Auth module tries to query default route /api/auth/user instead of /auth/user. I have to use this plugin because I need to pass current language (locale) as part of url (/lang/auth/user). Since we detect current language dynamically - we cannot hardcode it in nuxt config. 

### What is expected ?

Auth module should use the overridden routes from the plugin

### What is actually happening?

It seems that Auth module queries fetchUser before plugins/auth.js is applied. Because once the page in fully loaded/mounted then other routes works correctly (at least /auth/login)

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/auth-module/issues/c673">#c673</a>)</em></sub></div>

**Comment by elC0mpa** [comment]
I am having this problem too @NMFES, did you find a solution?

**Comment by NMFES** [comment]
@elC0mpa 
Nope. I left it for a while. And looks like they are not going to fix it.

**Comment by disjointed** [comment]
Yeah, in my project I've needed to be able to set endpoints, based on some ajax/axios results. After going trough like a dozen variants I've ended up using this kinda dumb, but working approach:
```
// nuxt.config.js
// ...
plugins: [
	'~plugins/axios',
],
axios: {
	baseURL: API_URL,
},
auth: {
	strategies: {
		jwt: {
			provider: 'laravel/jwt',
			url: API_URL,
		},
	},
	redirect: {
		logout: '/login',
	},
},
// ...
```
```
// ~/plugins/axios.js
export default async ({ $axios }) => {
	const client = await $axios.$get('endpoint');

	$axios.interceptors.request.use(
		(config) => {
			config.baseURL += `/clients/${client.id}`;

			if (config.url.endsWith('api/auth/login')) {
				config.url = 'auth/login';
			}

			if (config.url.endsWith('api/auth/logout')) {
				config.url = 'auth/logout';
			}

			if (config.url.endsWith('api/auth/refresh')) {
				config.url = 'auth/refresh';
			}

			if (config.url.endsWith('api/auth/user')) {
				config.url = 'auth/user';
			}

			return config;
		},
		error => Promise.reject(error),
	);
};
```

So I just fetch some data from API and then look at incoming requests URL and override it and add prefix. It's a really dumb solution, but for me it's the only one that works for this module.
BUT! This approach doesn't work in "Universal" site mode. It just doesn't ¯\_(ツ)_/¯ So, "SPA" mode only "t r i c k".

**Comment by elC0mpa** [comment]
thanks @disjointed I will try this later and let you know if this was useful for me
